reporters/reporter-default/src/main/java/org/radargun/reporting/html/IndexDocument.java

```diff
@@ -218,71 +72,91 @@ private void addToConfigs(Set<OriginalConfig> configs, int slave, String filenam
+            for (Map.Entry<Integer, Map<String, Properties>> entry : report.getNormalizedServiceConfigs().entrySet()) {
+               if (slaves.contains(entry.getKey()) && entry.getValue() != null) {
+                  normalized.addAll(entry.getValue().keySet());
+               }
+            }
+
+            Set<OriginalConfig> configs = new HashSet<>();
+            for (Map.Entry<Integer, Map<String, byte[]>> entry : report.getOriginalServiceConfig().entrySet()) {
+               if (slaves.contains(entry.getKey()) && entry.getValue() != null) {
+                  for (Map.Entry<String, byte[]> file : entry.getValue().entrySet()) {
+                     addToConfigs(configs, entry.getKey(), file.getKey(), file.getValue());
+                  }
+               }
+            }
+            this.configs.put(report, configs);
+
+            for (OriginalConfig config : configs) {
+               writeConfig(report.getCluster(), setup, config);
+            }
          }
       }
-      writer.write("</ul></li>\n");
    }
 
-   private String escape(String str) {
-      return str.replaceAll("<", "&lt;").replace(">", "&gt;");
+   /**
+    * The following methods are used in Freemarker templates
+    * e.g. method getPercentiles() can be used as getPercentiles() or percentiles in template
+    */
+
+   public Set<OriginalConfig> getConfigs(Report report) {
+      return configs.get(report);
    }
 
-   protected void writeTimelines(Collection<Report> reports) {
-      writeTag("h2", "Timelines");
-      write("<ul>\n");
-      for (Report report : reports) {
-         write(String.format("<li><a href=\"timeline_%s_%d.html\">%s on %s</a></li>",
-            report.getConfiguration().name, report.getCluster().getClusterIndex(), report.getConfiguration().name, report.getCluster()));
-      }
-      write("\n</ul>");
+   public String getFilename(String configName, String groupName, Cluster cluster, String config) {
+      return String.format("normalized_%s_%s_%d_%s.html",
+              configName, groupName, cluster.getClusterIndex(), config);
    }
 
-   protected void writeTests(Collection<String> testNames) {
-      writeTag("h2", "Tests");
-      write("<ul>\n");
-      for (String test : testNames) {
-         write(String.format("<li><a href=\"test_%s.html\">%s</a></li>", test, test));
-      }
-      write("\n</ul>");
+   public int getElementCounter() {
+      return elementCounter;
    }
 
-   protected void writeFooter() {
-      writer.write("<hr>");
-      writer.write("Generated on " + new Date() + " by RadarGun\nJDK: " +
-            System.getProperty("java.vm.name") + " (" + System.getProperty("java.vm.version") + ", " +
-            System.getProperty("java.vm.vendor") + ") OS: " + System.getProperty("os.name") + " (" +
-            System.getProperty("os.version") + ", " + System.getProperty("os.arch") + ")");
+   public void incElementCounter() {
```

Comment:
I think it makes sense to move element counter directly to template, as it's view-specific.